content/applications/inventory_and_mrp/inventory/shipping_receiving/advanced_operations_shipping/print_on_validation.rst

```diff
@@ -156,3 +160,188 @@ be printed as:
+details for specific third-party shipping carriers, complete the following setup:
+
+#. Tick the :guilabel:`Carrier Labels` checkbox in the :ref:`operation type settings
+   <inventory/shipping_receiving/print_setup>`.
+#. :doc:`Connect a printer <../../../../general/iot/devices/printer>` to Odoo's *IoT* app.
+#. :ref:`Assign the carrier label to the printer <inventory/shipping_receiving/assign-printer>`.
+#. Configure the shipping method's :ref:`label type <inventory/shipping_receiving/label-type>`.
+
+.. _inventory/shipping_receiving/assign-printer:
+
+Assign printer
+--------------
+
+Refer to the :doc:`Connect a printer <../../../../general/iot/devices/printer>` documentation for
+details on connecting a printer to Odoo's *IoT* app. After completing that, begin assigning the
+carrier label to the printer by navigating to :menuselection:`IoT app --> Devices` and selecting the
+desired printer.
+
+.. image:: print_on_validation/select-printer.png
+   :align: center
+   :alt: Show a list of IoT devices.
+
+In the printer configuration form, go to the :guilabel:`Printer Reports` tab to configure the types
+of documents the printer automatically prints. Click :guilabel:`Add a line` to open the
+:guilabel:`Add: Reports` pop-up window. In the :guilabel:`Search...` bar, type `Shipping` and select
+:guilabel:`Shipping Labels`.
+
+.. note::
+   The :guilabel:`Shipping Documents` report is for :ref:`export documents
+   <inventory/shipping_receiving/export-doc>`.
+
+.. image:: print_on_validation/printer-report.png
+   :align: center
+   :alt: Show carrier label report added to the *Printer Reports*.
+
+After adding the :guilabel:`Shipping Labels` report in the :guilabel:`Printer Reports` tab, ensure
+the :guilabel:`Report Type` matches the IoT-connected printer's type.
+
+- For laser printers, set the :guilabel:`Report Type` to :guilabel:`PDF`.
+- For Zebra printers, set the :guilabel:`Report Type` to :guilabel:`Text`.
+
+.. _inventory/shipping_receiving/label-type:
+
+Shipping carrier label type
+---------------------------
+
+Next, complete the setup for the :doc:`third party shipping connector
+<../setup_configuration/third_party_shipper>`. After that, go to :menuselection:`Inventory app -->
+Configuration --> Shipping Methods` and select the desired shipping method.
+
+On the shipping method configuration form, in the :guilabel:`[carrier name] Configuration` tab,
+ensure the :guilabel:`Label Format` matches the :ref:`report type assigned earlier
+<inventory/shipping_receiving/assign-printer>`:
+
+- For laser printers, set the :guilabel:`Label Format` to :guilabel:`PDF`.
+- For Zebra printers set the :guilabel:`Label Format` to :guilabel:`ZPL2`.
+
+.. image:: print_on_validation/label-type.png
+   :align: center
+   :alt: Show the *Label Type* field on FedEx's shipping method configuration page.
+
+Example carrier label
+---------------------
+
+After validating the delivery, the carrier label is generated in the chatter and printed using the
```

Comment:
```suggestion
After validating the delivery, the carrier label is generated in the chatter, and printed using the
```

Reply:
do you think it would be beneficial to add "...in the chatter of the delivery order,..." or do you think that is fairly self-explanatory? Totally not required, just a suggestion of something to consider during your revisions. 👍